F/yurik/2.0 primer

yurik42 opened the pull request:
BMG patches were applied to 2.0 tag
The code was compiled and "tested" with VS 2019

yurik42 committed:
Applied BMG patches

Changes to be committed:
modified:   src/openMVG/exif/sensor_width_database/ParseDatabase.hpp
modified:   src/openMVG/exif/sensor_width_database/sensor_width_camera_database.txt
modified:   src/openMVG/multiview/CMakeLists.txt
modified:   src/openMVG/robust_estimation/CMakeLists.txt
modified:   src/software/SfM/main_SfMInit_ImageListing.cpp

yurik42 committed:
Updated (BMG)

Changes to be committed:
modified:   src/openMVG/exif/sensor_width_database/ParseDatabase.hpp
modified:   src/openMVG/exif/sensor_width_database/sensor_width_camera_database.txt

Commit:
Merge branch 'develop' into f/yurik/2.0-primer

yurik42 committed:
Restored

Changes to be committed:
modified:   src/openMVG/exif/sensor_width_database/ParseDatabase.hpp

yurik42 commented on src/software/SfM/main_SfMInit_ImageListing.cpp:
```diff
@@ -127,6 +127,176 @@ std::pair<bool, Vec3> checkPriorWeightsString
+				}
+
+				// Compare the strings
+				if (0 == stricmp(view_base_filename.c_str(), image_name.c_str()))
```
Non-portable use of stricmp ?

mikec-bmg replied:
It looks like CPL uses a construct like below for a POSIX or Windows version of caseless compare:

```#  if defined(WIN32)
#    define STRCASECMP(a,b)         (stricmp(a,b))
#    define STRNCASECMP(a,b,n)      (strnicmp(a,b,n))
#  else
/** Alias for strcasecmp() */
#    define STRCASECMP(a,b)         (strcasecmp(a,b))
/** Alias for strncasecmp() */
#    define STRNCASECMP(a,b,n)      (strncasecmp(a,b,n))
#  endif
```
Of course, until we actually build for another platform this doesn't really matter, so probably no reason to change it until then.

mikec-bmg replied:
I have no idea why the 'Insert Code' made the font so big for some lines of that chunk of code.

yurik42 replied:
I added a macro definition
```
#ifdef _MSC_VER
#define strcasecmp(x, y) _stricmp(x, y)
#endif
```

yurik42 committed:
Added definition of strcasecmp() for MSVC compiler

Changes to be committed:
modified:   src/software/SfM/main_SfMInit_ImageListing.cpp

yurik42 committed:
Replaced stricmp with strcasecmp()

Changes to be committed:
modified:   src/software/SfM/main_SfMInit_ImageListing.cpp

vminor commented on src/software/SfM/main_SfMInit_ImageListing.cpp:
```diff
@@ -185,9 +366,12 @@ int main(int argc, char **argv)
       << "\t 0: ECEF (default)\n"
-      << "\t 1: UTM";
+      << "\t 1: UTM\n"
+      << "[-x|--exif_extended_lookup] Allow partial map on camera model name in EXIF information"
```
<< "[-x|--exif_extended_lookup] Allow partial map on camera model name in EXIF information\n"

yurik42 replied:
Are you suggesting to add '\n' at the end?
But the original line (190) did not have trailing '\n'

vminor replied:
yes, unless the [-x] parameter is part of the [-l]. It looks like we added the 3 new help suggestions, and we need a newline to make them look neat.

yurik42 committed:
Corrected "help" text of main_SfMInit_ImageListing

Changes to be committed:
modified:   ../src/software/SfM/main_SfMInit_ImageListing.cpp
